sql/inspect: fix LTREE[] handling in query bounds

spilchen opened the pull request:
INSPECT builds query bounds from primary key column types. When a table had an LTREE[] column, the bound datums could be interpreted as string[] by the query engine, causing errors like:
```
unsupported comparison operator: <ltree[]> >= <string[]>
```

This happened because the datum formatting omitted type information for array elements. For example, ['j.k.l']::LTREE[] was formatted as ARRAY['j.k.l'], which loses the cast. The fix marks the LTREE type as ambiguous so the formatter includes explicit type casts, avoiding mismatches in distributed queries.

While testing, I also added placeholder values to the logged query in inspect issues (previously only the query text was shown).

Informs: #155483
Closes: #156121
Epic: CRDB-55075

Release note: none

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156122)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @spilchen)



<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156122)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @rafiss and @spilchen)



<!-- Sent from Reviewable.io -->

rafiss reviewed (approved):
lgtm! thank you for tracking this down and fixing

mw5h reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

yuzefovich reviewed (commented):
Nice catch on `DLTree.AmbiguousFormat`! Do we plan to backport this change to 25.4? If not, we should probably backport the fix to `datum.go` file.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156122#-:-OclYIEo9Sav_jRLayqI:b-gq7yfu)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rafiss and @spilchen)



<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
No, the INSPECT changes won't be backported. I'll separate the changes into two commits so that we can backport just the change to `datum.go`.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/156122#-:-OclZSdeDVLjKHO2ogP4:bl9ybma)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rafiss and @spilchen)



<!-- Sent from Reviewable.io -->

spilchen committed:
sql/inspect: add placeholder values to logged queries

Previously, INSPECT only logged the query text without placeholder
values in issue reports, making debugging more difficult. This change
includes the actual placeholder values in the logged query output.

This also addes a test for INSPECT and LTREE[] handling.

Informs: #155483
Epic: CRDB-55075

Release note: none

spilchen committed:
sem/tree: mark LTREE as ambiguous type for proper casting

LTREE datums need explicit type casts when formatted to avoid being
misinterpreted as strings by the query engine. Previously, LTREE[]
arrays were formatted as ARRAY['j.k.l'] which loses type information,
causing the query engine to interpret them as string[] and resulting
in errors like:
```
unsupported comparison operator: <ltree[]> >= <string[]>
```

Marking LTREE as an ambiguous type ensures the formatter includes
explicit type casts (e.g., ARRAY['j.k.l']::LTREE[]), preventing type
mismatches in distributed queries.

Closes: #156121
Epic: CRDB-55075

Release note: none
